unityfiles/Assets/Scripts/Game/Collectables/Pickups/DoubleFireRateManager.cs

```diff
@@ -18,14 +22,14 @@ public class DoubleFireRateManager : MonoBehaviour {
     [SerializeField]
     private int secEffectDuration = 10;
 
-
-
     private int rollingID = 0;
     private GunController usedGunController = null;
     private int activeMultipliers = 0;
 
     private Dictionary<int, IEnumerator> activeCoroutines = new Dictionary<int, IEnumerator>();
 
+    public static event EventHandler<OnPickupDoubleFireRateArgs> OnPickupDoubleFireRateActive;
+    public static event EventHandler<OnPickupDoubleFireRateArgs> OnPickupDoubleFireRateExpired;
```

Comment:
som nevnt over med såpas spesifikke events er argumenter unødvendige. Enten ville jeg ha hat en event ie pickupState changedElns med de argunentene du har og teste på type/bool på andre siden, eller hvis du ønsker og ha spesifikke events trenger ikke disse og ha event args ie.
```
public static event EventHandler Onblalblabla;
Onblalblabla?.Invoke(this, EventArgs.Empty);
```

Reply:
Ahhh, takk for tips!!